libs/x-content/src/main/java/org/elasticsearch/xcontent/SerializableString.java

```diff
@@ -0,0 +1,33 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+package org.elasticsearch.xcontent;
+
+import java.util.function.Function;
+
+public class SerializableString {
+    private final String name;
+    private volatile Object serializedString = null;
+
+    public static SerializableString create(String name) {
```

Comment:
How about `of` instead of `create`? But just a very personal pref, pls ignore if you don't like